src/screencastwidget.c

```diff
@@ -268,24 +362,31 @@ void
 screen_cast_widget_set_app_id (ScreenCastWidget *widget,
                                const char *app_id)
 {
-  g_autofree char *heading = NULL;
+  g_autofree char *monitor_heading = NULL;
+  g_autofree char *window_heading = NULL;
 
   if (app_id && strcmp (app_id, "") != 0)
     {
       g_autofree char *id = NULL;
       g_autoptr(GAppInfo) info = NULL;
+      const gchar *format = _("Select %s to share with %s");
```

Comment:
Not sure this is that helpful for translators, compared to having one "Select monitor to share with %s" and one "Select window to share with %s".

Reply:
Recently I see a case where @afranke says we are told to not repeat a translate message when only some parts change. So the rule is we should do this way, @piotrdrag could know better too.

Reply:
I’m not sure what you are referring to but this doesn’t work here. “Monitor” and “window” can for instance have different genders in some languages and the verb may not to be adjusted to that. @jadahl is suggesting the appropriate strings.

Reply:
I see, was not aware of that. I had assumed different kind of words potentially requiring structuring the sentences differently, but then again I know very little about translation, so if you've been told to do it this way by experts then I have nothing more to say :)

Reply:
Just saw the reply above thus unresolving the issue until the strings are made completely separate.

Reply:
@jadahl did you miss my comment? I just said this is **not** correct.

Reply:
@afranke I posted before Github bothered to show me your comment, then saw in my inbox you had already replied :P

Reply:
Ok, let me rework it.

Reply:
@afranke fixed.

> My understanding is that Screen Share is the established term for this feature

Done too. 

So only @allanday concern about the message remains.

Here is other similar message (for screenshots) https://github.com/flatpak/xdg-desktop-portal-gtk/blob/master/src/screenshotdialog.c#L434

> Share this screenshot with %s?
> Share this screenshot with the requesting application?

We use right now on screencast:

> Select monitor to share with %s
>Select monitor to share with the requesting application
> Select window to share with %s
>Select window to share with the requesting application

but mockup will only use one message:

> <app id> is requesting permission to view your screen. Select which part to share

I think this last one is better, @allanday should I change that? Any further change you think we should apply?

![mockup](https://camo.githubusercontent.com/0ba57711a66e14859d80dff39504349f107e31ff/68747470733a2f2f6769746c61622e676e6f6d652e6f72672f5465616d732f44657369676e2f6f732d6d6f636b7570732f7261772f6d61737465722f706f7274616c732f73637265656e2d73686172696e672d72656d6f74652d6465736b746f702e706e67)

Reply:
@allanday I can change this if you want (and add the ellipsize), but right now I prefer to stick to the existing message for coherence. What do you think?